x-pack/plugins/observability_solution/slo/public/hooks/use_create_burn_rate_rule.tsx

```diff
@@ -26,29 +26,8 @@ export function useCreateRule<Params extends RuleTypeParams = never>() {
     theme,
   } = useKibana().services;
 
-  let loadingToastId = '';
-
-  const showLoadingToast = () => {
-    const loadingToast = toasts.addInfo({
-      title: toMountPoint(
-        <EuiFlexGroup justifyContent="center" alignItems="center">
-          <EuiFlexItem grow={false}>
-            {i18n.translate('xpack.slo.rules.createRule.loadingNotification.descriptionText', {
-              defaultMessage: 'Creating burn rate rule ...',
-            })}
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiLoadingSpinner size="s" />
-          </EuiFlexItem>
-        </EuiFlexGroup>,
-        { i18n: i18nStart, theme }
-      ),
-    });
-    loadingToastId = loadingToast.id;
-  };
-
-  const createRule = useMutation<
-    CreateRuleResponse<Params>,
+  return useMutation<
+    CreateRuleResponse<Params> & { loadingToastId?: string },
     Error,
     { rule: CreateRuleRequestBody<Params> }
   >(
```

Comment:
`loadingToastId` is not part of the Data response. it belongs to the context, which is the 4th type parameter. 


```suggestion
  return useMutation<
    CreateRuleResponse<Params>,
    Error,
    { rule: CreateRuleRequestBody<Params> },
    { loadingToastId: string }
  >(
```